SteamReview/SteamReview/Steam.cpp

```diff
@@ -0,0 +1,320 @@
+
+	std::cout << "Please enter Day ";
+	int InDay;
+	std::cin >> InDay;
+	InputValidate(InDay, 0, 31);
+
+	std::cout << "Please enter Month ";
+	int InMonth;
+	std::cin >> InMonth;
+	InputValidate(InMonth, 0, 12);
+
+	std::cout << "Please enter Year ";
+	int InYear;
+	std::cin >> InYear;
+	InputValidate(InYear, 1990, 2022);
+
+	//Creating a Game
+	return FGame(InGameName, InStudioName, InDay, InMonth, InYear);
+
+
+};
+
+//----------------------------------------------------------------
+bool FSteam::AddNewGameToCategory(const FGame InGame, const int UserInput)
+{
+	InputValidate(UserInput, 1, 2);
+	if (UserInput == 1)
+	{
+		//Choosing a category where the user going to add a game
+		std::system("cls");
+		std::cout << "------------------------------------------\nPlease Select the category where you want to add a game" << std::endl;
+		CategoryContainer.PrintCategories();
+
+		int CategoryChosen;
+		std::cin >> CategoryChosen;
+		InputValidate(CategoryChosen, 0, CategoryContainer.GetCurrentCategoryAmount());
+
+		//adding the game to category
+		if (CategoryContainer.AddGameToCategory(CategoryChosen - 1, InGame))
+		{
+			std::cout << "------------------------------------------\n" << std::endl;
+			std::cout << "The game is going to add to the category: " << CategoryContainer.GetCategory(CategoryChosen - 1).GetCategoryName() << std::endl;
+			return true;
+		}
+		else
+		{
+			std::cout << "------------------------------------------\n" << std::endl;
+			std::cout << "The game couldn't be added to the category: " << CategoryContainer.GetCategory(CategoryChosen - 1).GetCategoryName() << std::endl;
+			return false;
+		}
+	}
+	else
+	{
+		std::cout << "The game is going to add to the category: " << Uncategorized.GetCategoryName() << std::endl;
+		Uncategorized.AddGame(InGame);
+		return true;
+	}
+}
+
+//------------------------------------------------------------------------
+bool FSteam::AddNewCategory()
+{
+	//creating a new category
+	std::system("cls");
+	std::cout << "Please enter the name for the new category" << std::endl;
+	std::string CategoryName;
+	std::cin.ignore(500, '\n');
+	std::getline(std::cin, CategoryName);
+	if (CategoryContainer.AddCategory(CategoryName))
+	{
+		return true;
+	}
+	else
+	{
+		return false;
+	}
+	
+};
+
+//------------------------------------------------------------------------
+bool FSteam::DeleteACategory()
+{
+	if (CategoryContainer.IsEmpy())
+	{
+		std::cout << "You aren't saved any category yet..." << std::endl;
+		return false;
+	}
+	
+	else
+	{
+		//Deleting a category
+		std::system("cls");
+		CategoryContainer.PrintCategories();
+		std::cout << "-----------------------------------------------" << std::endl;
+
+		std::cout << "Selected the category to delete" << std::endl;
+		int CategorySelected;
+		std::cin >> CategorySelected;
+		InputValidate(CategorySelected, 0, CategoryContainer.GetCurrentCategoryAmount());
+		if (CategoryContainer.DeleteCategory(CategorySelected - 1))
```

Comment:
Here's another example of unnecessarily complicated return statements for a single condition.